src/Runner.php

```diff
@@ -2,87 +2,94 @@
 
 namespace Pantheon\Terminus;
 
-use Consolidation\AnnotatedCommand\AnnotatedCommandFactory;
 use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+use Consolidation\Log\Logger;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Robo\Robo;
+use Robo\Runner as RoboRunner;
 
 class Runner
 {
     /**
-     * @var Terminus
+     * @var Application
      */
     private $application;
     /**
-     * @var string
+     * @var \Robo\Runner
      */
-    private $commands_directory;
-    /**
-     * @var string
-     */
-    private $top_namespace = 'Pantheon\Terminus\Commands';
+    private $runner;
 
     /**
-     * Runner constructor
+     * Object constructor
      *
-     * @param array $options Options to configure the runner
+     * @param array $options Elements as follow:
+     *        Application application An instance of a Symfony Console Application
+     *        Config      config      A Terminus config instance
      */
-    public function __construct(array $options = [])
+    public function __construct(array $options = ['application' => null, 'config' => null,])
     {
-        $this->commands_directory = __DIR__ . '/Commands';
-        $this->application = new Terminus();
-        $this->configureApplication(new Config($options));
+        $this->application = $options['application'];
```

Comment:
Good news:  that's easy!

$container = new \League\Container\Container();
$app = new Terminus(...);
$input = new \Symfony\Component\Console\Input\ArgvInput($argv);
$output = new \Symfony\Component\Console\Output\ConsoleOutput();
Robo::configureContainer($container, $input, $output, $app);

Then you can pass the container to the runner constructor. First parameter is still your commandfile list; the second parameter is null and the third parameter is the constructor.